magefiles/build.go

```diff
@@ -0,0 +1,79 @@
+//go:build mage
+
+//nolint:wrapcheck
+package main
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"runtime"
+
+	"github.com/magefile/mage/sh"
+)
+
+const (
+	amd64 = "amd64"
+)
+
+var errUnknownTarget = errors.New("could not determine zig target")
+
+// Helper function to determine the Zig target triple based on OS and architecture.
+func getTargetTriple(goos, goarch string) (string, error) {
+	switch goos {
+	case "linux":
+		if goarch == amd64 {
+			return "x86_64-linux-gnu", nil
+		} else if goarch == "arm64" {
+			return "aarch64-linux-gnu", nil
+		}
+	case "darwin":
+		if goarch == amd64 {
+			return "x86_64-macos", nil
+		} else if goarch == "arm64" {
+			return "aarch64-macos", nil
+		}
+	case "windows":
+		if goarch == amd64 {
+			return "x86_64-windows-gnu", nil
+		}
+	}
+
+	return "", fmt.Errorf("%w: %s/%s", errUnknownTarget, goos, goarch)
+}
+
+// Build a Python wheel.
+func Build() error {
+	if err := sh.RunV("python3", "-mvenv", "env"); err != nil {
```

Comment:
it may be a better idea to set the venv to `build/.venv`
- `.venv` is the "usual" location for a local project's virtual environment
- stashing it under `build` means it wouldn't clash with the one a dev would be expected to create under the root of the project, as it would have a different purpose